pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -815,6 +815,22 @@ func (tmpl *Template) GetOutputs() *Outputs {
 	return nil
 }
 
+const (
+	// ExecutorStagingEmptyDir is the path of the emptydir which is used as a staging area to transfer a file between init/main container for script/resource templates
+	ExecutorStagingEmptyDir = "/argo/staging"
+	// ExecutorScriptSourcePath is the path which init will write the script source file to for script templates
+	executorScriptSourcePath = "/argo/staging/script"
+)
+
+func (tmpl *Template) GetScriptSourcePath() string {
+	extension := tmpl.Script.Extension
+	if len(extension) != 0 {
+		return executorScriptSourcePath + "." + tmpl.Script.Extension
```

Comment:
`Script` could be `nil` if I'm not mistaken, so that case should be handled as well

```suggestion
	if tmpl.Script != nil && len(tmpl.Script.Extension) != 0 {
		return executorScriptSourcePath + "." + tmpl.Script.Extension
```

Reply:
I was also thinking about this - my main issue was what should be returned if script is null. I've then decided that caller should ensure that script is not null.

But I guess returning just sourcePath should be fine

Reply:
Fixed

Reply:
We do get nil pointers with some frequency in the issues, and I'm still surprised that Go's type-checker doesn't prevent them (see also #12228).

Returning the default was my idea to maintain the correctness while still assuming that the caller is using it only on script templates.
It would be either that or return an error, which leads to more complicated code